crates/trie/parallel/src/root.rs

```diff
@@ -233,6 +233,15 @@ pub enum ParallelStateRootError {
     /// Provider error.
     #[error(transparent)]
     Provider(#[from] ProviderError),
+    /// Sparse Tree not revealed .
+    #[error("sparse tree should be revealed")]
+    SparseTreeNotRevealed,
+    /// Retention not enabled .
+    #[error("retention must be enabled")]
+    RetentionNotEnabled,
+    /// Sparse Tree not revealed and Retention not enabled .
+    #[error("Both root and trie_update should be visible")]
```

Comment:
```suggestion
    #[error("both root and trie update should be visible")]
```
nitpick, discouraged to use variable names in error messages